webpack/assets/javascripts/react_app/components/Pagination/Pagination.js

```diff
@@ -5,35 +5,108 @@ import { Paginator } from 'patternfly-react';
 import { getURI, getURIpage, getURIperPage, changeQuery, translatePagination } from './PaginationHelper';
 import './pagination.scss';
 
-const Pagination = ({ data }) => {
-  const urlPage = getURIpage();
-  const urlPerPage = getURIperPage();
+const Pagination = (props) => {
+  if (props.data) {
+    return erbPagination(props.data);
+  }
+  return paginationRow(props);
+}
+
+const erbPagination = (data) => {
+  const pagination = ({
+    page: getURIpage() || 1,
+    perPage: getURIperPage() || data.perPage,
+    perPageOptions: data.perPageOptions,
+  });
+
   const className = isEmpty(data.classNames) ? 'col-md-12' : `col-md-12 ${data.classNames.pagination_classes}`;
 
   return (
-      <Paginator
-        pagination={{
-          page: urlPage || 1,
-          perPage: urlPerPage || data.perPage,
-          perPageOptions: data.perPageOptions,
-        }}
-        viewType={data.viewType}
-        itemCount={data.itemCount}
-        onPageSet={page => changeQuery(getURI(), { page })}
-        onPerPageSelect={perPage => changeQuery(getURI(), { per_page: perPage })}
-        messages={translatePagination(Paginator.defaultProps.messages)}
-        className={className}
-      />
+    <Paginator
+      pagination={pagination}
+      viewType={data.viewType}
+      itemCount={data.itemCount}
+      onPageSet={page => changeQuery(getURI(), { page })}
+      onPerPageSelect={perPage => changeQuery(getURI(), { per_page: perPage })}
+      messages={translatePagination(Paginator.defaultProps.messages)}
+      className={className}
+    />
   );
+}
+
+
+const initPagination = (paginationFromProps) => {
+  const pagination = paginationFromProps || {};
+
+  const defaultPagination = {
+    page: 1,
+    perPage: 20,
+    perPageOptions: [5, 10, 15, 20, 25, 50],
```

Comment:
does it make sense to have all of these options as default propTypes?

Reply:
The defaults are shallow-merged, so having it in defaults does not really work if `{ page: 1, perPage: 10 }` is passed in.